cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/LogParserTransform.java

```diff
@@ -115,13 +120,35 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
         return;
       }
       output = parseRequest(logMatcher, S3_INDICES);
-    } else {
+    } else if (CLF_LOG.equals(config.logFormat)) {
       Matcher logMatcher = CLF_LOG_PATTERN.matcher(log);
       if (!logMatcher.matches() || logMatcher.groupCount() < CLF_REGEX_LENGTH) {
         LOG.debug("Couldn't parse log because the log did not match the CLF format. log: {}", log);
         return;
       }
       output = parseRequest(logMatcher, CLF_INDICES);
+    } else {
+      if (log.startsWith("#")) {
+        LOG.info("Log is a comment. Ignoring...");
```

Comment:
Should log to debug. Even trace for that matter. You're logging for every record right?